pkg/util/provider/machinecontroller/machine_util.go

```diff
@@ -241,12 +273,12 @@ func (c *controller) syncMachineNodeTemplates(ctx context.Context, machine *v1al
 		lastAppliedALT              v1alpha1.NodeTemplateSpec
 	)
 
-	node, err := c.nodeLister.Get(machine.Labels[v1alpha1.NodeLabelKey])
-	if err != nil && apierrors.IsNotFound(err) {
-		// Dont return error so that other steps can be executed.
-		return machineutils.LongRetry, nil
-	}
+	node, err := c.nodeLister.Get(getNodeName(machine))
 	if err != nil {
+		if apierrors.IsNotFound(err) {
+			// Don't return error so that other steps can be executed.
+			return machineutils.LongRetry, nil
+		}
 		klog.Errorf("Error occurred while trying to fetch node object - err: %s", err)
 		return machineutils.LongRetry, err
```

Comment:
Just to make it consistent across the changed methods.
```suggestion
		return machineutils.ShortRetry, err
```